scylla-cdc-base/src/main/java/com/scylladb/cdc/transport/WorkerTransport.java

```diff
@@ -13,5 +13,13 @@
     Map<TaskId, TaskState> getTaskStates(Set<TaskId> tasks);
     void setState(TaskId task, TaskState newState);
     void moveStateToNextWindow(TaskId task, TaskState newState);
-    boolean shouldStop();
+
+    /**
+     * @deprecated Use {@link Worker.stop()} instead")
```

Comment:
I don't understand how is Worker.stop a replacement for shouldStop? stop sets internal state and returns/exposes nothing while shouldStop changes no internal state and exposes whether it was requested through transport to stop the worker.

Reply:
Replacement does not imply "equal semantics". Yes, the client needs to actually call stop to use it. But at some point it also has to decide to return true here, so it is not like there isn't a natural switch-over point where you can either set an internal state to return here, or call the worker and ask it to stop. 
To quote myself from above, the call to "stop()" is potentially faster and also serves as a sync-point. You call it, and when returned, the connector is not running anymore.